compiler/optimizations/remoteValueForwarding.cpp

```diff
@@ -460,6 +462,306 @@ static void serializeAtCallSites(FnSymbol* fn,  ArgSymbol* arg,
   }
 }
 
+/*  
+ *  This function handles deserialization of AggregateTypes that have ref
+ *  fields. Those fields need to point to data that is local for RVF to be
+ *  meaningful. However, if the data is created inside the deserializer of that
+ *  aggregate type, it'll be freed at the end of deserialization. Therefore, we
+ *  hoist the deserialization of that data out of the main deserializer and into
+ *  the `on` function. This is repeated recursively. The transformation is
+ *  complicated and looks like the following for a single-ref-field record:
```

Comment:
I am seeing two cases:

- A promotion that has an array view, and
- A promotion that has another promotion as a field.

Interesting point about ref tuples. Just as a context: they were causing a problem back when we used to use tuples for `forall ... zip` loops. I have hit similar problems in my first attempt to solve this because of that. But we decided that we should get rid of tuples there, so we were able to punt this issue with promotion.